File: algorithm/Python3_version/leetcode5441.making-file-names-unique.py
class Solution:
    def getFolderNames(self, names: List[str]) -> List[str]:
        m = {}
        ans = [""]*len(names)
        for i in range(len(names)):
            if names[i] not in m:
                m[names[i]] = 1
                ans[i] = names[i]
            else:
                num = m[names[i]]
                while names[i]+'('+str(num)+')' in m:
                    num += 1
                ans[i] = names[i]+'('+str(num)+')'
                m[names[i]] = num+1
                m[ans[i]] = 1
        return ans

File: algorithm/leetcode5441.making-file-names-unique.cpp
class Solution {
public:
    vector<string> getFolderNames(vector<string>& names) {
    	unordered_map<string,int> m;
    	vector<string> ans(names.size());
        int num;
    	for(int i = 0; i < names.size(); ++i)
    	{
    		if(!m.count(names[i]))//没有出现过
    		{
    			m[names[i]]++;//计数+1
    			ans[i] = names[i];//答案不变
    		}
    		else
    		{	//出现过该单词
                num = m[names[i]];//读取当前应该的计数
    			while(m.count(names[i]+"("+to_string(num)+")"))//出现过了
                    num++;//数字增加，直到没有出现过
                ans[i] = names[i]+"("+to_string(num)+")";//写入答案
    			m[names[i]] = num+1;//更新计数
    			m[ans[i]]++;
    		}
    	}
    	return ans;
    }
};